test_conformance/integer_ops/test_integer_dot_product.cpp

```diff
@@ -0,0 +1,395 @@
+    typedef typename std::make_unsigned<DstType>::type UDstType;
+
+    // dot testing:
+    result |= test_case_dot<UDstType, USrcType, USrcType, N>(
+        deviceID, context, queue, num_elements, false, false);
+    result |= test_case_dot<SDstType, SSrcType, SSrcType, N>(
+        deviceID, context, queue, num_elements, false, false);
+    result |= test_case_dot<SDstType, USrcType, SSrcType, N>(
+        deviceID, context, queue, num_elements, false, false);
+    result |= test_case_dot<SDstType, SSrcType, USrcType, N>(
+        deviceID, context, queue, num_elements, false, false);
+
+    // dot_acc_sat testing:
+    result |= test_case_dot<UDstType, USrcType, USrcType, N>(
+        deviceID, context, queue, num_elements, false, true);
+    result |= test_case_dot<SDstType, SSrcType, SSrcType, N>(
+        deviceID, context, queue, num_elements, false, true);
+    result |= test_case_dot<SDstType, USrcType, SSrcType, N>(
+        deviceID, context, queue, num_elements, false, true);
+    result |= test_case_dot<SDstType, SSrcType, USrcType, N>(
+        deviceID, context, queue, num_elements, false, true);
+
+    return result;
+}
+
+template <typename SrcType, typename DstType, size_t N>
+static int test_vectype_packed(cl_device_id deviceID, cl_context context,
+                               cl_command_queue queue, int num_elements)
+{
+    int result = TEST_PASS;
+
+    typedef typename std::make_signed<SrcType>::type SSrcType;
+    typedef typename std::make_signed<DstType>::type SDstType;
+
+    typedef typename std::make_unsigned<SrcType>::type USrcType;
+    typedef typename std::make_unsigned<DstType>::type UDstType;
+
+    // packed dot testing:
+    result |= test_case_dot<UDstType, USrcType, USrcType, N>(
+        deviceID, context, queue, num_elements, true, false);
+    result |= test_case_dot<SDstType, SSrcType, SSrcType, N>(
+        deviceID, context, queue, num_elements, true, false);
+    result |= test_case_dot<SDstType, USrcType, SSrcType, N>(
+        deviceID, context, queue, num_elements, true, false);
+    result |= test_case_dot<SDstType, SSrcType, USrcType, N>(
+        deviceID, context, queue, num_elements, true, false);
+
+    // packed dot_acc_sat testing:
+    result |= test_case_dot<UDstType, USrcType, USrcType, N>(
+        deviceID, context, queue, num_elements, true, true);
+    result |= test_case_dot<SDstType, SSrcType, SSrcType, N>(
+        deviceID, context, queue, num_elements, true, true);
+    result |= test_case_dot<SDstType, USrcType, SSrcType, N>(
+        deviceID, context, queue, num_elements, true, true);
+    result |= test_case_dot<SDstType, SSrcType, USrcType, N>(
+        deviceID, context, queue, num_elements, true, true);
+
+    return result;
+}
+
+int test_integer_dot_product(cl_device_id deviceID, cl_context context,
+                             cl_command_queue queue, int num_elements)
+{
+    if (is_extension_available(deviceID, "cl_khr_integer_dot_product"))
```

Comment:
You could invert the condition and early return to not have to indent all the code.

Reply:
Done.